### PR TITLE
Essence Update

### PR DESCRIPTION
**Pull request by Pukami**
Just made essence look nicer using the bloom effect, and made it so that they can time out properly.

### CODE REVIEW

File: src/game_objects/essence.cpp [hunk 1]
```diff
@@ -141,4 +144,30 @@ void Essence::random_push() {
...
+	anim_player.play();
+}
+
+void Essence::setup_timeout() {
+	// Make essence disappear after a while.
+	int time_out = ESSENCE_TIME_OUT + (int)Util::random(-1000, 1000); // Slightly diff time outs
+	Timer::get()->do_after(
+		std::chrono::milliseconds(time_out),
+		[&]() mutable {
+		events::dungeon::network_collision_event(id, id);
```
Comment (richard1) [inline src/game_objects/essence.cpp]: Is this intentional? (passing in `id` twice to this call)

**Review by richard1 — COMMENTED** [review]
Looks good overall, just one question.  Also it looks like there are a lot of tabs throughout, could you replace those with spaces?

File: src/game_objects/essence.cpp (hunk 1 above)
Comment (Pukami) [reply src/game_objects/essence.cpp]: Yes. It's currently the easiest way to send an animation call to the client.

File: src/game_objects/essence.cpp [hunk 2]
```diff
@@ -141,4 +144,30 @@ void Essence::random_push() {
     rigidbody->setActivationState(true);
     rigidbody->applyForce(btVector3(vel.x, 0, vel.y), btVector3(0, 0, 0));
     rigidbody->setDamping(0.9, 0);
+}
+
+void Essence::pickup_anim() {
+	// Add "pick up" animation
+
```
Comment (jytang) [inline src/game_objects/essence.cpp]: remove this newline

**Review by jytang — APPROVED** [review]

**Pull request merged**